controllers/replica_set_controller.go

```diff
@@ -306,6 +307,12 @@ func (r *ReplicaSetReconciler) ensureTLSResources(mdb mdbv1.MongoDBCommunity) er
 		if err := ensureTLSSecret(r.client, mdb); err != nil {
 			return fmt.Errorf("could not ensure TLS secret: %s", err)
 		}
+		if mdb.Spec.GetAgentAuthMode() == "X509" {
```

Comment:
Very nice that the changes in the reconciler are minimal! ❤️

Reply:
Was one of the objectives 😄